src/history_be/CanisterHistory.mo

```diff
@@ -95,14 +95,30 @@ module {
     /// Returns `true` if the sync is successful, and `false` if there is an ongoing sync.
     /// If there is an ongoing sync, then new one is not starting.
     public func sync() : async* Bool {
-      if (sync_ongoing) return false;
+      let ?(ic, args) = schedule_sync_call() else return false;
```

Comment:
Why don't we inline the function `schedule_sync_call()` here? It doesn't seem to do anything interesting.